### PR TITLE
Merge remote-tracking branch 'local-5.6/ps-5.6-4529' into ps-5.7-4529

### PR DESCRIPTION
**Pull request by george-lorch**
(cherry picked from commit f4472e3cd8787cbc7ea7e62067fc114f6ab3e2c0)

### CODE REVIEW

**Comment by george-lorch** [comment]
https://ps.cd.percona.com/view/8.0/job/percona-server-8.0-param/105/

File: mysql-test/suite/rocksdb/r/index_merge_rocksdb2.result [hunk 1]
```diff
@@ -535,11 +536,10 @@ key(cola), key(colb)
 INSERT INTO t1 VALUES ('foo','bar', 'ZZ'),('fuz','baz', 'ZZ');
 OPTIMIZE TABLE t1;
 Table	Op	Msg_type	Msg_text
-test.t1	optimize	note	Table does not support optimize, doing recreate + analyze instead
 test.t1	optimize	status	OK
 select count(*) from t1;
 count(*)
-8704
+526
```
Comment (laurynas-biveinis) [inline mysql-test/suite/rocksdb/r/index_merge_rocksdb2.result:542]: Say what?

Comment (george-lorch) [reply mysql-test/suite/rocksdb/r/index_merge_rocksdb2.result:542]: Actually, neither 526 nor 8704 make any sense:

> # BUG#16166
> CREATE TABLE t1 (
> cola char(3) not null, colb char(3) not null,  filler char(200),
> key(cola), key(colb)
> );
> INSERT INTO t1 VALUES ('foo','bar', 'ZZ'),('fuz','baz', 'ZZ');
> OPTIMIZE TABLE t1;
> Table	Op	Msg_type	Msg_text
> test.t1	optimize	status	OK
> select count(\*) from t1;
> count(\*)
> 526

To me it looks like the result should be "2" all day long from any engine, but, both InnoDB and MyISAM return "8704" and MyRocks is returning "526" :-\

Comment (george-lorch) [reply mysql-test/suite/rocksdb/r/index_merge_rocksdb2.result:542]: Nevermind, this is just from the result, the test itself disables the query log and inserts a bunch of rows.  So yeah, MyRocks is doing something wrong here, but for some reason only in 8.0 (or it's possible this is a new test bit in 8.0).

Comment (george-lorch) [reply mysql-test/suite/rocksdb/r/index_merge_rocksdb2.result:542]: This is actually another utf8mb4 secondary key bug https://jira.percona.com/browse/PS-5024

**Review by laurynas-biveinis — APPROVED** [review]

**Pull request merged**